src/components/mainGoal/CheckDelete.js

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { useNavigate } from "react-router-dom";
+import PropTypes from "prop-types";
+import axios from "axios";
+
+export default function CheckDelete({ onClick, target }) {
+  const navigate = useNavigate();
+
+  const handleConfirmButtonClick = async () => {
+    const res = await deleteGoal();
+
+    if (res?.result === "ok") {
+      onClick(false);
+      navigate("/main");
+      return;
+    }
+
+    onClick(false);
+  };
```

Comment:
제가 잘못 이해한 것이 아니라면 아마 둘 중에 하나의 onClick 값으로는 true 값이 들어가야 할 것 같네요
```suggestion
    if (res?.result === "ok") {
      onClick(true);
      navigate("/main");
      return;
    }

    onClick(false);
  };
```

Reply:
안녕하세요 수진님! 전체적인 리팩토링을 하느라 코멘트가 늦어지게 되어 죄송합니다.
현재는 약간 바뀌기는 했는데요, 성공/실패 시를 기준으로 modal이 닫히거나 다른 액션을 할 수 있도록 분기를 만들었습니다. 프론트엔드 에러 분기처리 형식 통일에 대해서 한 번 논의한 후 작업하려고 둘 다 modal이 바로 닫히는 걸로 세팅해두었습니다.